pkg/visor/logstore/logstore.go

```diff
@@ -0,0 +1,86 @@
+package logstore
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// LogRealLineKey is a key in the log enry that denote real log line number
+// in the total log (not limited by capacity of runtime log store)
+const LogRealLineKey = "real_line"
+
+// Store is in-memory log store that returns all logs as a single string
+type Store interface {
+	// get logs returns stored logs and the number of log entries overwritten
+	// due to insufficient capacity.
+	// returned number n means that n log entries have been dropped and the oldest
+	// log entry is (n+1)th
+	GetLogs() ([]*logrus.Entry, int64)
+	// get logs as a single string
+	GetLogStr() (string, error)
+}
+
+// MakeStore returns a new store that will hold up to max entries,
+// overwriting the oldest entry when over the capacity
+// returned hook should be registered in logrus master logger to
+// store log entries
+func MakeStore(max int) (Store, logrus.Hook) {
+	entries := make([]*logrus.Entry, max)
+	formatter := &logrus.JSONFormatter{}
+	store := &store{max: int64(max), entries: entries, formatter: formatter}
+	return store, store
+}
+
+type store struct {
+	max       int64
+	n         int64
+	entries   []*logrus.Entry
+	formatter logrus.Formatter
+}
```

Comment:
What is **n** exactly? In the GetLogs function use as number of log lines, isn't?

Reply:
It's the number of logs processed and the number of the next log entry to process.
It's used to calculate the index in the limited capacity of the store and also could be used to append actual log entry number to the response. I refactored names a bit, n is probably too generic